pandas/core/arrays/datetimelike.py

```diff
@@ -1435,6 +1435,50 @@ def max(self, axis=None, skipna=True, *args, **kwargs):
         # Don't have to worry about NA `result`, since no NA went in.
         return self._box_func(result)
 
+    def mean(self, axis=None, skipna=True):
```

Comment:
Do we want to add an `axis` argument here? 
I know we have already added in some places, but also not added in other places, so probably something we should discuss.

I would personally try to not add too much overhead in additional useless kwargs just for matching signatures, *if* they can be avoided.

Reply:
IIRC, this lets us hook into `np.mean`, but we should verify exactly which kwargs are necessary.

Reply:
Yes, that's why we did it in the past (but for that we also need the additional kwargs). But, for EAs, we could also decide to implement the array function protocol to achieve this goal, which might obviate the need to add the additional kwargs

Reply:
@jbrockmendel i think its worthile here to allow calling ``np.mean``

Reply:
don'e we also need out=? meaning we should just accept kwargs like we do for Series.mean